### DISCUSSION TITLE
Interchangeable head

### DISCUSSION THREAD

**Discussion by Sylensky** [xpDIY/3DPlacer#3] [Ideas]
Hi,
Awesome work you did there with the 3DPlacer, absolutely loving it!
Do you have future plans to make the printer head interchangeable?
From the firmware perspective it would be awesome to have a pnp function included in the 3d printers firmware but thats a topic for another long story..

How successfull in terms of accurracy have you been so far with 0201 components?

**Comment by xpDIY** [comment]
Hi,
Thanks, I am looking for an easy way to switch, currently it can interchange between 3D printing and pnp with 2 screws, and require to adapt 2 fan wires, to go for full interchangeable, existing printer need to adapt also the hot end cable and temperature sensor and Bowden tube, I plan to first remove the need for 2 screws, then later full switch, but keep both options available. 
For the placement, I got around 0.1mm accuracy without modifications on firmware of 3d printer, for me the 0201 could deviate a bit, but got pull back to pad during reflow, then I change the minimal step in Marlin fw(most of 3D printer combine several step to 1), then I got around 0.02 to 0.04mm) which is much better, so yes, fw itself also has room for adaptation regarding pnp functionality, I will make a video to show also about this topic.
